pre_commit_hooks/string_fixer.py

```diff
@@ -32,7 +33,14 @@ def get_line_offsets_by_line_no(src):
 
 
 def fix_strings(filename):
-    contents = io.open(filename).read()
+    try:
+        contents = io.open(filename).read()
+    except Exception:
```

Comment:
Hmmm, actually the stdlib *does* support byte tokenizing:

## python2
- `tokenize.generate_tokens` supports both bytes and text

## python3
- `tokenize.tokenize` supports only bytes, the first token will be an `ENCODING` token which should be used to re-encode the result
- each token will have `str` (text) contents

So this is workable without using bytes throughout I think

I wonder if I could improve [tokenize-rt](https://github.com/asottile/tokenize-rt) to handle bytes to normalize these differences (and then use it here)